### ISSUE TITLE
Feature idea: lock field value

### ISSUE THREAD

**Issue by DavidOliver** [brendo/default_event_values#12]
I'm thinking it would be good to have a checkbox labelled `Lock field value` which would prevent a field's value from being changed via a frontend event. Like the `Members: Lock Role` filter does for member entry updates.

To do this at present, we need to put a field's current value into a parameter, don't we?

**Comment by brendo** [comment]
No, it doesn't have to be a parameter, just a string will do :)

**Comment by DavidOliver** [comment]
But using strings doesn't allow us to lock the field's value to whatever it already is in the database, does it? This is for updating existing entries rather than creating new ones.

**Comment by brendo** [comment]
Ah sorry, I see what you mean now. Yeah, that would set the field to be the
value of the string, not maintain the existing value.

**Comment by DavidOliver** [comment]
Okeydoke. In my current project/use case, it's unlikely that DOM hackers would guess the field names as these fields are never exposed via any forms. But it would seem to be a good option to have? If you think so, I'll add it to the list of my things to have a go at.

**Comment by brendo** [comment]
Definitely worthwhile doing, it's a great idea! Let me know if you need any
support with this :)

**Comment by DavidOliver** [comment]
Cool. Thanks!

**Comment by DavidOliver** [comment]
A couple more thoughts:

Or should this be handled in the core via a `Disallow updating via Events` checkbox on every field? But then the option wouldn't be event-specific.

Is locking a field away from editing via an event in-scope for this “default values” extension? If the option were added, should the extension be renamed?

**Comment by brendo** [comment]
In a loose interpretation, default could mean new value or default to an
existing value. I'm happy for it to live in this extension :)

**Comment by DavidOliver** [comment]
Okay - thanks for confirming.
